validator/project_validator_test.go

```diff
@@ -1299,38 +1301,52 @@ func TestCheckProjectSemantics(t *testing.T) {
 }
 
 func TestEnsureHasNecessaryProjectFields(t *testing.T) {
-	Convey("When ensuring necessary project fields are set, ensure that", t, func() {
-		Convey("projects validate all necessary fields exist", func() {
-			Convey("an error should be thrown if the batch_time field is "+
-				"set to a negative value", func() {
-				project := &model.Project{
-					Enabled:     true,
-					Identifier:  "identifier",
-					Owner:       "owner",
-					Repo:        "repo",
-					Branch:      "branch",
-					DisplayName: "test",
-					RepoKind:    "github",
-					BatchTime:   -10,
-				}
-				So(ensureHasNecessaryProjectFields(project),
-					ShouldNotResemble, []ValidationError{})
-				So(len(ensureHasNecessaryProjectFields(project)),
-					ShouldEqual, 1)
-			})
-			Convey("an error should be thrown if the command type "+
-				"field is invalid", func() {
-				project := &model.Project{
-					BatchTime:   10,
-					CommandType: "random",
-				}
-				So(ensureHasNecessaryProjectFields(project),
-					ShouldNotResemble, []ValidationError{})
-				So(len(ensureHasNecessaryProjectFields(project)),
-					ShouldEqual, 1)
-			})
-		})
-	})
+	assert := assert.New(t)
```

Comment:
I would use testify's suite functionality rather than this scoping thing (which is clever, but isn't an idiom that I've seen used in go, also because go is GC'd and doesn't have deterministic destruction, it's less useful than it'd be in C++)